src/Textarea.jsx

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+import cx from 'classnames';
+import autosize from 'autosize';
+
+/**
+ * @module Textarea
+ */
+class Textarea extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			value: props.value || '',
+		};
+		this.onChange = this.onChange.bind(this);
+	}
+
+	onChange(e) {
+		this.setState({
+			value: e.target.value,
+		});
+	}
+
+	componentDidMount() {
+		if (this.props.autoheight){
```

Comment:
`componentDidMount` is definitely preferred if it works since it will only get called once for the lifecycle of the element, but if the autosizer loses its reference to the DOM element when React re-renders (without unmounting) the component, then you'll have to call it in the render method - I think just play around with it and see if it ever loses the `autosize` functionality in unexpected ways